### ISSUE TITLE
missing_panics_doc is issued for debug_assert!

### ISSUE THREAD

**Issue by Licenser** [rust-lang/rust-clippy#6970]
Lint name:

missing_panics_doc

I tried this code:

```rust
    /// merge the other `ids` into this one
    pub fn merge(&mut self, ids: &TrackedEventIds) {
        debug_assert!(self.source_id == ids.source_id, "incompatible source ids");
        debug_assert!(self.stream_id == ids.stream_id, "incompatible stream ids");
        // self.track_ids(ids.min_event_id, ids.max_event_id); this line is commented out to guarantee that the panic is really in debug_assert!
    }```

I expected to see this happen: `debug_assert!` not be considered a panic since it is for debug compilations

Instead, this happened: This function is considered panicking by clippy

### Meta

- `cargo clippy -V`: `clippy 0.1.51 (2fd73fa 2021-03-23)`
- `rustc -Vv`:
  ```
rustc 1.51.0 (2fd73fabe 2021-03-23)
binary: rustc
commit-hash: 2fd73fabe469357a12c2c974c140f67e7cdd76d0
commit-date: 2021-03-23
host: x86_64-unknown-linux-gnu
release: 1.51.0
LLVM version: 11.0.1
  ```

**Comment by lopopolo** [comment]
I think I previously reported this here when this change landed on nightly: https://github.com/rust-lang/rust-clippy/issues/6739.

Sorry for not doing a more thorough investigation.

This false positive triggered in 3+ crates that I maintain.

Is doing crater-style runs for new clippy lints to test for false positives something that can be considered before new lints make it to stable?

**Comment by Licenser** [comment]
Ja I saw your bug report but the comment suggested that it was related to something else :). It seems it got fixed by @Y-Nak. Thanks for the quick fix!
